Client/Coordinators/Library/DownloadsCoordinator.swift

```diff
@@ -0,0 +1,64 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol DownloadsNavigationHandler: AnyObject {
+    /// Handles the possibile navigations for a file.
+    /// The source view is the view used to display a popover for the share controller.
+    func handleFile(_ file: DownloadedFile, sourceView: UIView)
+}
+
+class DownloadsCoordinator: BaseCoordinator, DownloadsNavigationHandler, UIDocumentInteractionControllerDelegate {
+    // MARK: - Properties
+
+    private weak var parentCoordinator: LibraryCoordinatorDelegate?
+
+    // MARK: - Initializers
+
+    init(
+        router: Router,
+        parentCoordinator: LibraryCoordinatorDelegate?
+    ) {
+        self.parentCoordinator = parentCoordinator
+        super.init(router: router)
+    }
+
+    // MARK: - DownloadsNavigationHandler
+
+    func handleFile(_ file: DownloadedFile, sourceView: UIView) {
+        if file.mimeType == MIMEType.Calendar {
```

Comment:
I'm not sure if we want this type of checks in coordinator. @OrlaM what do you think?

Reply:
@FilippoZazzeroni This should probably be 2 functions on the DownloadsNavigationHandler, one to trigger the UIDocumentInteractionController flow and the other to trigger the share sheet flow. 
That way the business logic of deciding which flow to take based on the document type can be kept out of the coordinator.

Reply:
ok perfect thank you, i'll move this implementation out of the coordinator